2.6/root/usr/share/container-scripts/mongodb/base-functions.sh

```diff
@@ -0,0 +1,76 @@
+# This file contains functions for checking mongod state and manipulating config file
+
+# Default constants
+export MAX_ATTEMPTS=90
+export SLEEP_TIME=2
+
+# Wait_mongo waits until the mongo server is up/down
+# $1 - "UP" or "DOWN" - to specify for what to wait
+# $2 - host where to connect (localhost by default)
+function wait_mongo() {
+  local operation
+  operation=-eq
+  if [[ "${1:-}" == "DOWN" || "${1:-}" == "down" ]]; then
+    operation=-ne
+  fi
+
+  local mongo_cmd
+  mongo_cmd="mongo admin --host ${2:-localhost} --port $port "
+
+  local i
+  for i in $(seq $MAX_ATTEMPTS); do
+    echo "=> ${2:-} Waiting for MongoDB daemon ${1:-}"
+    set +e
+    ${mongo_cmd} --eval "quit()" &>/dev/null
+    local status=$?
+    set -e
+    if [ ${status} ${operation} 0 ]; then
+      echo "=> MongoDB daemon is ${1:-}"
+      return 0
+    fi
+    sleep ${SLEEP_TIME}
+  done
+  echo "=> Giving up: MongoDB daemon is not ${1:-}!"
+  return 1
+}
+
+# Get value for option in file
+# $1 - option name
+# $2 - path to config file
+function get_option() {
+  [[ -z "${1:-}" || -z "${2:-}" || ! -r "${2:-}" ]] && return 1
+
+  grep "^\s*${1}" ${2} | sed -r -e "s|^\s*${1}\s*=\s*||"
+}
+
+# Get port number from config file
+# $1 - path to config file
+function get_port() {
+  [[ -z "${1:-}" || ! -r "${1:-}" ]] && return 1
+
+  if grep '^\s*port' $1 &>/dev/null; then
+    grep '^\s*port' $1 | sed -r -e 's|^\s*port\s*=\s*(\d*)|\1|'
+  elif grep '^\s*configsvr' $1 &>/dev/null; then
+    echo 27019
+  elif grep '^\s*shardsvr' $1 &>/dev/null; then
+    echo 27018
+  else
+    echo ""
+  fi
+}
+
+# Change value for config option in configuration file
+# $1 - option name
+# $2 - new value
+# $3 - path to config file
+function update_option() {
+  [[ -z "${1:-}" || -z "${2:-}" || -z "${3:-}" || ! -r "${3:-}" ]] && return 1
+
+  # Delete old option from config file
+  sed -r -e "/^\s*$1/d" $3 > ${HOME}/.tmp.conf
+  cat ${HOME}/.tmp.conf > $3
+  rm ${HOME}/.tmp.conf
+
+  # Add new option into config file
+  echo "$1 = $2" >> $3
```

Comment:
Yes it is. But this can't influence the next run of this function - `sed -r -e "/^\s*$1/d" $3 > ${HOME}/.tmp.conf` completely rewrite this file. And "in case of the error" the container will stop, so it doesn't matter.

Also it is not possible by this one-line (I wanted to use first) -  `sed -i` uses temp file in directory where the result file is. And mongodb linux user has no right to create this file in `/etc`, so therefore there is a need to create this temporary file in HOME.